x/tx/signing/direct_aux/direct_aux.go

```diff
@@ -0,0 +1,111 @@
+package direct_aux
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cosmos/cosmos-proto/anyutil"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoregistry"
+
+	signingv1beta1 "cosmossdk.io/api/cosmos/tx/signing/v1beta1"
+	txv1beta1 "cosmossdk.io/api/cosmos/tx/v1beta1"
+	"cosmossdk.io/x/tx/signing"
+)
+
+// SignModeHandler is the SIGN_MODE_DIRECT_AUX implementation of signing.SignModeHandler.
+type SignModeHandler struct {
+	signersContext *signing.GetSignersContext
+	fileResolver   protodesc.Resolver
+	typeResolver   protoregistry.MessageTypeResolver
+}
+
+// SignModeHandlerOptions are the options for the SignModeHandler.
+type SignModeHandlerOptions struct {
+	// FileResolver is the protodesc.Resolver to use for resolving proto files when unpacking any messages.
+	FileResolver protodesc.Resolver
+
+	// TypeResolver is the protoregistry.MessageTypeResolver to use for resolving proto types when unpacking any messages.
+	TypeResolver protoregistry.MessageTypeResolver
+
+	// SignersContext is the signing.GetSignersContext to use for getting signers.
+	SignersContext *signing.GetSignersContext
+}
+
+// NewSignModeHandler returns a new SignModeHandler.
+func NewSignModeHandler(options SignModeHandlerOptions) SignModeHandler {
+	h := SignModeHandler{}
+
+	if options.FileResolver == nil {
+		h.fileResolver = protoregistry.GlobalFiles
+	} else {
+		h.fileResolver = options.FileResolver
+	}
+
+	if options.TypeResolver == nil {
+		h.typeResolver = protoregistry.GlobalTypes
+	} else {
+		h.typeResolver = options.TypeResolver
+	}
+
+	if options.SignersContext == nil {
+		h.signersContext = signing.NewGetSignersContext(signing.GetSignersOptions{})
```

Comment:
We should be able to pass the file resolver into these options. I think it just requires us changing the options struct to use the correct interface

Reply:
Can you explain a bit more what you mean?

Here a user could set GetSignersContext.protoFiles to whatever they want to by constructing it themselves and passing it in.

Reply:
Yeah, but if they pass in `FileResolver` but not `GetSignersContext`, the `GetSignersContext` should be constructed with that `FileResolver`. Generally `FileResolver` and `TypeResolver` should be propagated to any component that needs them because the probuf types in a scope should all be the same.

Reply:
Ok this will be a refactor to GetSignersContext then, right?

Reply:
Yes a bit. We probably need a new interface that extends FileResolver